src/EditorFeatures/Core/SymbolSearch/RoslynSymbolResult.cs

```diff
@@ -1,22 +1,73 @@
-﻿using Microsoft.CodeAnalysis.FindUsages;
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Language.Intellisense.SymbolSearch;
 using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Adornments;
 
 namespace Microsoft.CodeAnalysis.Editor.SymbolSearch
 {
-    internal class RoslynSymbolResult : SymbolSearchResult, IResultInLocalFile
+    internal class RoslynSymbolResult : SymbolSearchResult, IResultInLocalFile, IResultHasClassifiedDefinition, IResultHasClassifiedContext //, IResultOpensWithCustomAction
     {
         private DefinitionItem definition;
+        private SourceReferenceItem reference;
+        private RoslynSymbolSource source;
+        private static ISymbolOriginDefinition LocalSymbolOrigin = null;
 
-        public RoslynSymbolResult(DefinitionItem definition)
+        public RoslynSymbolResult(RoslynSymbolSource source, DefinitionItem definition)
         {
+            this.source = source;
             this.definition = definition;
+            Owner = source;
+            Name = definition.ToString();
+            Origin = GetLocalOrigin();
+            ClassifiedDefinition = new ClassifiedTextElement(definition.NameDisplayParts.Select(n => new ClassifiedTextRun(n.Style.ToString(), n.Text)));
+            ClassifiedContext = new ClassifiedTextElement(definition.DisplayParts.Select(n => new ClassifiedTextRun(n.Style.ToString(), n.Text)));
+            var sampleLocation = definition.SourceSpans.First(); // TODO: handle all locations
+            PersistentSpan = source.ServiceProvider.PersistentSpanFactory.Create(sampleLocation.Document.FilePath, AsSpan(sampleLocation.SourceSpan), SpanTrackingMode.EdgeInclusive);
```

Comment:
how does this work.  what if the file has changed since the search started?  It feels like thsi is wrong and will need to map from teh original doc location to wherever it is now.